### PR TITLE
[KOGITO-2986] Remove controller package dependency in tests

### PR DESCRIPTION
**Pull request by radtriste**
https://issues.redhat.com/browse/KOGITO-2986

Test should not depend on `controller` package directly but should make usage of `infrastructure` package

Many thanks for submiting your Pull Request :heart:! 

Please make sure that your PR meets the following requirements:

- [x] You have read the [contributors' guide](https://github.com/kiegroup/kogito-cloud-operator/blob/master/README.md#contributing-to-the-kogito-operator)
- [x] Pull Request title is properly formatted: `[KOGITO-XYZ] Subject`
- [x] Pull Request contains link to the JIRA issue
- [x] Pull Request contains description of the issue
- [x] Pull Request does not include fixes for issues other than the main ticket
- [ ] Your feature/bug fix has a unit test that verifies it
- [ ] You've tested the new feature/bug fix in an actual OpenShift cluster

### CODE REVIEW

**Review by ricardozanini — COMMENTED** [review]
Part of this I've already did in #602, but instead of using strings, I used the actual group version from the given package. Can you please use the same approach? So we can merge yours first since it's simpler.

**Comment by ricardozanini** [comment]
The OLM error can be ignored for now, I'll take a look once I finish #602 .

**Comment by radtriste** [comment]
https://github.com/operator-framework/community-operators/pull/2491 is created for the OLM error

**Comment by radtriste** [comment]
udpated

> Part of this I've already did in #602, but instead of using strings, I used the actual group version from the given package. Can you please use the same approach? So we can merge yours first since it's simpler.

updated

File: pkg/infrastructure/keycloak.go [hunk 1]
```diff
@@ -14,10 +14,24 @@
...
 package infrastructure
 
-import "github.com/kiegroup/kogito-cloud-operator/pkg/client"
+import (
+	"github.com/keycloak/keycloak-operator/pkg/apis/keycloak/v1alpha1"
+	"github.com/kiegroup/kogito-cloud-operator/pkg/client"
+)
 
 const (
 	keycloakServerGroup = "keycloak.org"
```
Comment (ricardozanini) [inline pkg/infrastructure/keycloak.go:23]:
```suggestion
	keycloakServerGroup = v1alpha1.SchemeGroupVersion.Group
```

**Review by ricardozanini — APPROVED** [review]
Looks great! Thanks!

**Review by ricardozanini — APPROVED** [review]

**Review by Sgitario — APPROVED** [review]

**Review by sutaakar — APPROVED** [review]

**Pull request merged**